File: LeetCode/2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.java
class Solution {
    private int mapValue(int num, int[] mapping) {
        if (num == 0) {
            return mapping[0];
        }
        int ans = 0, place = 1;
        while (num > 0) {
            ans += mapping[num % 10] * place;
            place *= 10;
            num /= 10;
        }
        return ans;
    }
    public int[] sortJumbled(int[] mapping, int[] nums) {
        int n = nums.length;
        int[][] mappedNums = new int[n][2];

        for (int i = 0; i < n; i++) {
            mappedNums[i][0] = mapValue(nums[i], mapping);
            mappedNums[i][1] = nums[i];
        }

        Arrays.sort(mappedNums, (a, b) -> Integer.compare(a[0], b[0]));
        int[] sortedNums = new int[n];
        for (int i = 0; i < n; i++) {
            sortedNums[i] = mappedNums[i][1];
        }

        return sortedNums;
    }
}

File: LeetCode/2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.cpp
class Solution {
public:
    int mapValue(int num, const vector<int>& mapping) {
        if (num == 0) return mapping[0];
        int mappedVal = 0, place = 1;
        while (num > 0) {
            mappedVal += mapping[num % 10] * place;
            num /= 10;
            place *= 10;
        }
        return mappedVal;
    }
    
    vector<int> sortJumbled(vector<int>& mapping, vector<int>& nums) {
        vector<pair<int, int>> mappedNums;
        
        for (int num : nums) {
            mappedNums.push_back({mapValue(num, mapping), num});
        }
        
        stable_sort(mappedNums.begin(), mappedNums.end(), [](const pair<int, int>& a, const pair<int, int>& b) {
            return a.first < b.first;
        });
        
        vector<int> sortedNums;
        for (const auto& p : mappedNums) {
            sortedNums.push_back(p.second);
        }
        
        return sortedNums;
    }
};